### PR TITLE
Rc migration web apis

### PR DESCRIPTION
**Pull request by roshanroutray**

### CODE REVIEW

File: src/Microsoft.ServiceFabric.Actors.Migration/Extensions/KvsActorStateProviderExtensionHelper.cs [hunk 1]
```diff
@@ -176,7 +176,7 @@ internal static async Task ResumeWritesAsync(this KvsActorStateProvider statePro
             }
         }
 
-        internal static bool GetWriteStatusAsync(this KvsActorStateProvider stateProvider)
+        internal static bool GetRejectWriteState(this KvsActorStateProvider stateProvider)
```
Comment (pakunapamsft) [inline src/Microsoft.ServiceFabric.Actors.Migration/Extensions/KvsActorStateProviderExtensionHelper.cs:179]: TODO: Consider caching this value.

File: src/Microsoft.ServiceFabric.Actors.Migration/KVStoRCMigrationActorStateProvider.cs [hunk 2]
```diff
@@ -356,6 +358,149 @@ internal IReliableStateManagerReplica2 GetStateManager()
...
+                long lastAppliedSN, startSN, lastSN;
+
+                cancellationToken.ThrowIfCancellationRequested();
+                using (var tx = this.GetStateManager().CreateTransaction())
+                {
+                    migrationStartTimeUtcValue = await metaDataDictionary.TryGetValueAsync(tx, MigrationConstants.MigrationStartTimeUtcKey);
+                    currentMigrationPhaseStartTimeUtcValue = await metaDataDictionary.TryGetValueAsync(tx, MigrationConstants.CurrentMigrationPhaseStartTimeUtcKey);
+                }
+
+                DateTime.TryParse(Encoding.ASCII.GetString(migrationStartTimeUtcValue.Value), out DateTime migrationStartTimeUtc);
```
Comment (pakunapamsft) [inline src/Microsoft.ServiceFabric.Actors.Migration/KVStoRCMigrationActorStateProvider.cs:406]: Either proceed based on return value of TryParse or use simply use Parse

File: src/Microsoft.ServiceFabric.Actors/Runtime/ActorManager.cs [hunk 3]
```diff
@@ -1086,6 +1102,17 @@ private void ThrowIfClosed()
             }
         }
 
+        private void ThrowIfMigrationInProgress()
+        {
+            bool rejectWriteState = this.actorService.GetRejectWriteState();
```
Comment (pakunapamsft) [inline src/Microsoft.ServiceFabric.Actors/Runtime/ActorManager.cs:1107]: The check has to be (Is_KVS_Migration_Source and RejectWrites_Set) or (Is_RC_MigrationTarget and MigrationPhase_Is_Not_Completed) instead?

**Review by pakunapamsft — CHANGES_REQUESTED** [review]

File: src/Microsoft.ServiceFabric.Actors.Migration/Extensions/KvsActorStateProviderExtensionHelper.cs (hunk 1 above)
Comment (roshanroutray) [reply src/Microsoft.ServiceFabric.Actors.Migration/Extensions/KvsActorStateProviderExtensionHelper.cs:179]: Added TODO comment

File: src/Microsoft.ServiceFabric.Actors/Runtime/ActorManager.cs (hunk 3 above)
Comment (roshanroutray) [reply src/Microsoft.ServiceFabric.Actors/Runtime/ActorManager.cs:1107]: Already doing this check in Microsoft.ServiceFabric.Actors.Migration.Utility.GetRejectWriteState.

File: src/Microsoft.ServiceFabric.Actors.Migration/KVStoRCMigrationActorStateProvider.cs (hunk 2 above)
Comment (roshanroutray) [reply src/Microsoft.ServiceFabric.Actors.Migration/KVStoRCMigrationActorStateProvider.cs]: Done

**Pull request merged**